app-backend/batch/src/main/scala/com/azavea/rf/batch/export/spark/Export.scala

```diff
@@ -1,90 +1,135 @@
 package com.azavea.rf.batch.export.spark
 
-import io.circe.parser._
-import com.azavea.rf.datamodel._
+import java.util.UUID
+
+import cats.data.Validated._
+import cats.implicits._
 import com.azavea.rf.batch._
-import com.azavea.rf.batch.util._
-import com.azavea.rf.batch.export._
+import com.azavea.rf.batch.ast._
 import com.azavea.rf.batch.dropbox._
+import com.azavea.rf.batch.export._
+import com.azavea.rf.batch.util._
 import com.azavea.rf.batch.util.conf._
-
-import geotrellis.proj4.LatLng
+import com.azavea.rf.common.InterpreterException
+import com.azavea.rf.datamodel._
+import com.azavea.rf.tool.ast.MapAlgebraAST
+import com.azavea.rf.tool.params.EvalParams
+import com.dropbox.core.v2.DbxClientV2
+import com.dropbox.core.v2.files.{CreateFolderErrorException, WriteMode}
+import com.typesafe.scalalogging.LazyLogging
+import geotrellis.proj4.{CRS, LatLng}
 import geotrellis.raster._
-import geotrellis.raster.io._
 import geotrellis.raster.histogram.Histogram
-import geotrellis.raster.io.geotiff.{GeoTiff, MultibandGeoTiff}
+import geotrellis.raster.io._
+import geotrellis.raster.io.geotiff.{GeoTiff, MultibandGeoTiff, SinglebandGeoTiff}
 import geotrellis.spark._
 import geotrellis.spark.io._
-import geotrellis.spark.tiling._
-import geotrellis.spark.io.hadoop._
 import geotrellis.spark.io.file._
+import geotrellis.spark.io.hadoop._
 import geotrellis.spark.io.s3._
-
-import com.typesafe.scalalogging.LazyLogging
+import geotrellis.spark.tiling._
+import geotrellis.vector.MultiPolygon
+import _root_.io.circe.parser._
 import org.apache.hadoop.fs.Path
 import org.apache.spark._
+import org.apache.spark.rdd.RDD
 import spray.json.DefaultJsonProtocol._
-import cats.implicits._
-import com.dropbox.core.v2.files.{CreateFolderErrorException, WriteMode}
 
-import java.util.UUID
+// --- //
 
 object Export extends SparkJob with Config with LazyLogging {
+
   val jobName = "Export"
 
+  def astExport(
+    ed: ExportDefinition,
+    ast: MapAlgebraAST,
+    params: EvalParams,
+    ingestLocs: Map[UUID, String],
+    conf: HadoopConfiguration
+  )(implicit sc: SparkContext): Unit = {
+    interpretRDD(ast, params.sources, ed.input.resolution, ingestLocs) match {
+      case Invalid(errs) => throw new InterpreterException(errs)
+      case Valid(rdd) => {
+
+        val mt: MapKeyTransform = rdd.metadata.layout.mapTransform
+        val crs: CRS = rdd.metadata.crs
+
+        if(!ed.output.stitch) {
+          /* Create GeoTiffs and output them */
+          val singles: RDD[(SpatialKey, SinglebandGeoTiff)] =
+            rdd.map({ case (key, tile) => (key, SinglebandGeoTiff(tile, mt(key), crs)) })
+
+          writeGeoTiffs[Tile, SinglebandGeoTiff](singles, ed, conf)
```

Comment:
Is there any way to get around having to manually specify the type params? You'd think it would be able to infer from the type of `singles`.